### ISSUE TITLE
[koa-shopify-auth] verifyToken perhaps should use GET not POST

### ISSUE THREAD

**Issue by kavika-1** [Shopify/quilt#1450]
## Overview

Was stepping through the redirect paths and noticed verify-token.ts fetch always returns 400, not 401 or 200.  The error body indicates "missing parameter".  Perhaps the intent was for `https://${session.shop}/admin/metafields.json` to be a GET (Method.Get)?

**Comment by TheMallen** [comment]
@ragalie does this make sense to you?

**Comment by ragalie** [comment]
Yeah this makes sense to me.

Am I reading this right that we're making an extra API call on every request though? Seems not very performant 😬

**Comment by tanema** [comment]
Yes, I think that is common for validating authentication in our normal flow. In other tools I use a metafield query as well. This should probably be a GET.

**Comment by ragalie** [comment]
:+1: yeah it's a simple way to do that check. And if someone wants to do something more complicated (but also more performant) like wrap their existing calls to the API and redirect if the token is bad, there's nothing stopping them.
